File: C++/007. Reverse Integer.cpp
class Solution {
public:
    int reverse(int x) {
        long long result = 0;
        int num;
        int f;
        if(x==0)
        {
            return 0;
        }
        else if(x>0)
        {
            f=1;
        }
        else
        {
            f=-1;
        }
        x*=f;
        for(int i=1;;i++)
        {
            num = x%10;
            result = 10*result+num;
            x = (x-num)/10;
            if(x==0)
            {
                break;
            }
        }
        result *= f;
        return (result<INT_MIN || result>INT_MAX)? 0:(int)result;
    }
};

File: python/007. Reverse Integer.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x < 0: sign = -1
        else: sign = 1
        res = 0
        x = abs(x)
        while x > 0:
            res = 10 * res + (x % 10)
            x //= 10
        res *= sign
        if res > (2 ** 31 - 1) or res < (-1 * 2 ** 31):
            return 0
        else:
            return res
